nixos/modules/services/mail/opendkim.nix

```diff
@@ -91,6 +101,22 @@ in {
 
   config = mkIf cfg.enable {
 
+    services.opendkim.settings = {
+      Background = false;
+      Syslog = true;
+
+      # Set this to null so we don't have to check for this key existing below
+      KeyTable = mkOptionDefault null;
+
+      # TODO: Maybe use mkIfNotDefault from https://github.com/NixOS/nixpkgs/pull/75584
+      Socket = mkDefault cfg.socket;
+      # These settings wouldn't have any effect when KeyTable is set
+      Domain = mkIf (cfg.settings.KeyTable == null) (mkDefault cfg.domains);
+      Selector = mkIf (cfg.settings.KeyTable == null) (mkDefault cfg.selector);
```

Comment:
As someone unfamiliar with running this service would it make sense to document these options are only applied if `KeyTable` is set, or does the man file (which I have not read) make it clear enough?

Reply:
Sounds good, did that